### ISSUE TITLE
Ensure that Orchard spends respect `-orchardanchorconfirmations` option

### ISSUE THREAD

**Issue by str4d** [zcash/zcash#5644]
We are going to implement #5570 as just selecting the most recent anchor, and the fix this once https://github.com/zcash/incrementalmerkletree/issues/22 is implemented.

**Comment by daira** [comment]
I do not think we should have a separate config option for Orchard. As far as I can see, there's no reason to use a different anchor policy for different shielded pools. There's a long-standing issue (#1614) to choose less recent anchors for JoinSplits and Sapling Spends as well.

**Comment by str4d** [comment]
I'd added an Orchard-specific config in order to try and get Orchard to at least be consistent, but was foiled by `incrementalmerkletree` limitations. Given that delay, I think we should just consider `-orchardanchorconfirmations` to be ignored, and when we fix this change the flag to `-anchorconfirmations`.

**Comment by nuttycom** [comment]
The value provided for `-orchardanchorconfirmations` interacts with the `minconf` parameter in that we must select an anchor recent enough to contain any notes selected given the provided `minconf` depth. However, `minconf` cannot directly override `-orchardanchorconfirmations` because it is possible to select a value >100 for `minconf`, in which case we may not be able to obtain a checkpoint in the incremental merkle tree corresponding to the selected anchor. As such, the strategy we have to implement is to limit note selection to notes with at least `minconf` confirmations, as we have now, but select the anchor according to `min(anchorConfirmations, minConf)` so that we are guaranteed to have an anchor that contains any notes that might be selected.
